src/bits/mod.rs

```diff
@@ -71,5 +71,5 @@ pub trait Serializable: Sized {
     /// buffer has enough capacity to fit instances of this trait.
     ///
     /// [`serialize`]: Self::serialize
-    fn deserialize(buf: GenericArray<u8, Self::Size>) -> Self;
+    fn deserialize(buf: &GenericArray<u8, Self::Size>) -> Self;
```

Comment:
We had a few cases where we used `GenericArray::clone_from_slice` in order to produce this value.  That involved a copy, which wasn't necessary, whereas `GenericArray::from_slice` doesn't.  But that returns a reference.

Reply:
that's right. I don't mind changing it this way, just saying that for places where I believe currently deserialize is "free" (bitarray) it will come with a cost of cloning.

Reply:
Yeah, though I realized that this is what we really want in those cases.  We want to own the copy of whatever came off the wire.  I also discovered that those cases can rely on the type being `Copy`, so it's cheap.